p5js/KNNClassification/KNNClassification_PoseNet/sketch.js

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2018 ml5
+//
+// This software is released under the MIT License.
+// https://opensource.org/licenses/MIT
+
+/* ===
+ml5 Example
+KNN Classification on Webcam Images with poseNet. Built with p5.js
+=== */
+let video;
+// Create a KNN classifier
+const knnClassifier = ml5.KNNClassifier();
+let poseNet;
+let poses = [];
+
+function setup() {
+  const canvas = createCanvas(640, 480);
+  canvas.parent('videoContainer');
+  video = createCapture(VIDEO);
+  video.size(width, height);
+
+  // Create the UI buttons
+  createButtons();
+
+  // Create a new poseNet method with a single detection
+  poseNet = ml5.poseNet(video, modelReady);
+  // This sets up an event that fills the global variable "poses"
+  // with an array every time new poses are detected
+  poseNet.on('pose', function(results) {
+    poses = results;
+  });
+  // Hide the video element, and just show the canvas
+  video.hide();
+}
+
+function draw() {
+  image(video, 0, 0, width, height);
+
+  // We can call both functions to draw all keypoints and the skeletons
+  drawKeypoints();
+  drawSkeleton();
+}
+
+function modelReady(){
+  select('#status').html('model Loaded')
+}
+
+// Add the current frame from the video to the classifier
+function addExample(label) {
+  // Convert poses results to a 2d array [[score0, x0, y0],...,[score16, x16, y16]]
+  const poseArray = poses[0].pose.keypoints.map(p => [p.score, p.position.x, p.position.y]);
+
+  // Add an example with a label to the classifier
+  knnClassifier.addExample(poseArray, label);
+  updateCounts();
+}
+
+// Predict the current frame.
+function classify() {
+  // Get the total number of labels from knnClassifier
+  const numLabels = knnClassifier.getNumLabels();
+  if (numLabels <= 0) {
+    console.error('There is no examples in any label');
+    return;
+  }
+  // Convert poses results to a 2d array [[score0, x0, y0],...,[score16, x16, y16]]
+  const poseArray = poses[0].pose.keypoints.map(p => [p.score, p.position.x, p.position.y]);
```

Comment:
Noting that this is something we may want to simplify later as discussed in https://github.com/ml5js/ml5-library/issues/245 and https://github.com/ml5js/ml5-library/issues/255. It's also an interesting question as to whether or not to include `p.score` and/or what to do with keypoints that have a very low confidence score.